tests/tlm/test_properties.py

```diff
@@ -1,3 +1,204 @@
+import asyncio
+from typing import Any, Dict, List, Union
+
 import pytest
 
-# Empy file - no code to test yet, just so CI passes
+from cleanlab_studio.internal.constants import (
+    _VALID_TLM_MODELS,
+    _VALID_TLM_QUALITY_PRESETS,
+)
+from cleanlab_studio.studio.trustworthy_language_model import TLM
+from tests.tlm.test_get_trustworthiness_score import is_trustworthiness_score
+from tests.tlm.test_prompt import is_tlm_response
+
+_VALID_TLM_MODELS = ["gpt-4o-mini", "gpt-3.5-turbo-16k", "gpt-4", "gpt-4o", "claude-3-haiku"]
+
+
+def _test_log(response: Dict[str, Any], options: Dict[str, Any]) -> None:
+    """Tests the log dictionary in the response based on the options dictionary."""
+    if "log" in options.keys():
+        assert isinstance(response["log"], dict)
+        if "perplexity" in options["log"]:
+            assert (
+                isinstance(response["log"]["perplexity"], float)
+                or response["log"]["perplexity"] is None
+            )
+        if "explanation" in options["log"]:
+            assert isinstance(response["log"]["explanation"], str)
+
+
+def _test_log_batch(responses: Dict[str, Any], options: Dict[str, Any]) -> None:
+    """Tests the log dictionary in the batch response based on the options dictionary."""
+    for response in responses:
+        if response is not None:
+            _test_log(response, options)
+
+
+def _test_prompt_response(response, options):
+    """Property tests the responses of a prompt based on the options dictionary and returned responses."""
+    assert response is not None
+    assert is_tlm_response(response)
+    _test_log(response, options)
+
+
+def _test_batch_prompt_response(responses, options):
+    """Property tests the responses of a batch prompt based on the options dictionary and returned responses."""
+    assert responses is not None
+    assert isinstance(responses, list)
```

Comment:
We also have to test that every item in this is `is_tlm_response()` right?

Reply:
Ah nvm I see it [here](https://github.com/cleanlab/cleanlab-studio/blob/8f42ba3741ac1daade546bb80b352c541bb60f18/tests/tlm/test_properties.py#L135) -- would it make more sense to shift it in this function tho?

Reply:
This is different depending on if its a try_ function (where response could be None) and a regular function, that is why I kept it out of this. 

With Jay's change this test will change overall so I think we can just ignore for now